base/array.jl

```diff
@@ -1251,10 +1251,11 @@ julia> findnext(A,3)
 function findnext(A, start::Integer)
     for i = start:length(A)
         if A[i] != 0
-            return i
```

Comment:
I agree with @KristofferC that the `findnext` example isn't the best use-case for this feature because a `return` suffices; I just included it for the sake of show-casing the working patch.

@yuyichao , yes that is indeed equivalent, and it's probably a matter of taste which is easier to understand.

IMHO, in a language where there is no distinction between statements and expressions, it is a bit wasteful and arguably unexpected (e.g. when compared to your `if`/`else` example) not to be able to specify a value for a loop expression. That's why I personally like this feature.